src/Magnum/Math/Quaternion.h

```diff
@@ -733,6 +740,57 @@ template<class T> Matrix3x3<T> Quaternion<T>::toMatrix() const {
     };
 }
 
+/* Algorithm from:
+   https://github.com/qt/qtbase/blob/84056cdca11c63ddb9fac3d95fcfa952adaa6ea1/src/gui/math3d/qquaternion.cpp#L507 */
+template<class T> Vector3<T> Quaternion<T>::toRotation() const {
+    Vector3<T> euler;
+
+    T xp = vector().x();
+    T yp = vector().y();
+    T zp = vector().z();
+    T wp = scalar();
+
+    T xx = xp*xp;
+    T xy = xp*yp;
+    T xz = xp*zp;
+    T xw = xp*wp;
+    T yy = yp*yp;
+    T yz = yp*zp;
+    T yw = yp*wp;
+    T zz = zp*zp;
+    T zw = zp*wp;
+
+    const T lengthSquared = xx + yy + zz + wp*wp;
+    if(!TypeTraits<T>::equalsZero(lengthSquared - 1.0f, 100) &&
+        !TypeTraits<T>::equalsZero(lengthSquared, 100)) {
+        xx /= lengthSquared;
+        xy /= lengthSquared;
+        xz /= lengthSquared;
+        xw /= lengthSquared;
+        yy /= lengthSquared;
+        yz /= lengthSquared;
+        yw /= lengthSquared;
+        zz /= lengthSquared;
+        zw /= lengthSquared;
+    }
+
+    euler.x() = std::asin(-2.0f*(yz - xw));
+    if(euler.x() < Constants<T>::pi()) {
+        if(euler.x() > -Constants<T>::pi()) {
+            euler.y() = std::atan2(2.0f*(xz + yw), 1.0f - 2.0f*(xx + yy));
+            euler.z() = std::atan2(2.0f*(xy + zw), 1.0f - 2.0f*(xx + zz));
+        } else {
+            euler.z() = 0.0f;
+            euler.y() = -std::atan2(-2.0f*(xy - zw), 1.0f - 2.0f*(yy + zz));
+        }
+    } else {
+        euler.z() = 0.0f;
+        euler.y() = std::atan2(-2.0f*(xy - zw), 1.0f - 2.0f*(yy + zz));
```

Comment:
All the literals need to be `T(0.0)` and `T(2.0)` respectively to avoid compiler warnings when using the double-precision `Quaterniond`.